### ISSUE TITLE
Target Adaptive Loss

### ISSUE THREAD

**Issue by kshen3778** [DIAL-RPI/PIPO-FAN#5]
Where can we find the implementation of the target adaptive loss mentioned in the paper?
Also what is the difference between the concave_dps and concave_dps_w? It seems the Softmax, attention, and pooling only exists in concave_dps_w.py?

**Comment by xifang001** [comment]
Hi, Shen,

The target adaptive loss can be seen at the 302nd  in train_sf_partial.py. More general version may be delivered in future. Concave_dps corresponds to the PIPO part, the outputs of PIPO are five feature maps, which helps we learn the weights for different scales. On the other hand, concave_dps_w correspond to PIPO + FAN, the output of is a probability map. Thus softmax and attention are added in concave_dps_w. That is to say, concave_dps (PIPO) is part of concave_dps_w (PIPO+FAN).

Best,
Xi

**Comment by kshen3778** [comment]
Thank you. Sorry, forgot to reply earlier. Also, in the paper it says " For each epoch, we randomly select three continuous slices containing target organ label from all the CT training volumes" is that the number of input channels to the model - the input size would be (batch size, 3, width, height)? Would the target also be the masks for those three slices or only the middle slice?

**Comment by xifang001** [comment]
Yes. The input size is set to be (batch size, 3, width, height) to enhance the context information. It's also named as 2.5D network. Only the mask of middle slice is used for training.

**Comment by kshen3778** [comment]
That makes sense. Thanks!
